src/thing_creature.c

```diff
@@ -5292,16 +5292,32 @@ int claim_neutral_creatures_in_sight(struct Thing *creatng, struct Coord3d *pos,
         {
             if (is_neutral_thing(thing) && line_of_sight_3d(&thing->mappos, pos))
             {
-				if (!creature_is_kept_in_custody(creatng))
-				{
-				    change_creature_owner(thing, creatng->owner);
+                // Unless the relevant classic bug is enabled,
+                // neutral creatures in custody (prison/torture) can only be claimed by the player who holds it captive
+                // and neutral creatures can not be claimed by creatures in custody.
+                if ((gameadd.classic_bugs_flags & ClscBug_PassiveNeutrals)
+                    || (get_room_creature_works_in(thing)->owner == creatng->owner && !creature_is_kept_in_custody(creatng))
+                    || !(creature_is_kept_in_custody(thing) || creature_is_kept_in_custody(creatng)))
+                {
+                    change_creature_owner(thing, creatng->owner);
                     mark_creature_joined_dungeon(thing);
                     if (is_my_player_number(thing->owner))
                     {
-                        output_message(SMsg_CreaturesJoinedYou, MESSAGE_DELAY_CRTR_JOINED , true);
+                        output_message(SMsg_CreaturesJoinedYou, MESSAGE_DELAY_CRTR_JOINED, true);
                     }
                     n++;
-				}
+                }
+                // Unless the relevant classic bug is enabled,
+                // neutral creatures in custody (prison/torture) can only be claimed by the player who holds it captive
+                // and neutral creatures can not be claimed by creatures in custody.
+                if ((gameadd.classic_bugs_flags & ClscBug_PassiveNeutrals)
```

Comment:
are these 2 ifs identical? don't see what the second one is for

Reply:
this looks like a merge error, I'll fix